### PR TITLE
fix: update deprecated utcnow, utcfromtimestamp

### LINKED ISSUE #2285: Deprecated utcnow, utcfromtimestamp should be removed
Python 3.12 [deprecates](https://docs.python.org/3.12/whatsnew/3.12.html#deprecated) `datetime.utcnow` and `datetime.utcfromtimestamp`. Their uses should be removed from the project.

### PR DESCRIPTION
**Pull request by salty-horse**
Fixes #2285 🦕

The docs should be regenerated, since they mention the `EPOCH` constants. I'm not sure how to do that :(

### CODE REVIEW

**Comment by vchudnov-g** [comment]
Thanks for the PR! Could you link to the docs to which you refer that need updating?

**Comment by salty-horse** [comment]
The epy-generated docs for the two files I changed, which mention the old value of `EPOCH`:
```
docs/epy/googleapiclient.channel-pysrc.html
docs/epy/googleapiclient.discovery_cache.file_cache-module.html
docs/epy/googleapiclient.discovery_cache.file_cache-pysrc.html
```

**Review by parthea — APPROVED** [review]

**Comment by parthea** [comment]
I filed https://github.com/googleapis/google-api-python-client/issues/2300 to track the issue with the outdated docs

File: googleapiclient/discovery_cache/file_cache.py [hunk 1]
```diff
@@ -46,7 +46,7 @@
 LOGGER = logging.getLogger(__name__)
```
Comment (Scoring100) [inline googleapiclient/discovery_cache/file_cache.py:46]: ~~_**
**_~~

File: tests/test_discovery.py [hunk 2]
```diff
@@ -2193,7 +2193,7 @@ def _dummy_token(self):
         client_id = "some_client_id"
         client_secret = "cOuDdkfjxxnv+"
```
Comment (Scoring100) [inline tests/test_discovery.py:2194]:
```suggestion
         client_secret = "cOuDdkfjxxnv+"``
```

File: tests/test_discovery.py [hunk 3]
```diff
@@ -2193,7 +2193,7 @@ def _dummy_token(self):
         client_id = "some_client_id"
```
Comment (Scoring100) [inline tests/test_discovery.py:2193]: -

**Review by Scoring100 — CHANGES_REQUESTED** [review]
Delete @briandorsey

**Pull request merged**